### PR TITLE
Created Ball class

### PR DESCRIPTION
**Pull request by ValeriyM250**
Created enum with different colors
Implemented method String getRandomColor() in class ColorSupplier
Created a Lottery class with getRandomBall() method

### CODE REVIEW

File: src/main/java/core/basesyntax/Application.java [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 public class Application {
     public static void main(String[] args) {
-        // create three balls using class Lottery and print information about them in console
+
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Application.java:5]: remove empty line

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 2]
```diff
@@ -1,7 +1,10 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
     public String getRandomColor() {
-        return null;
+        int index = new Random().nextInt(Color.values().length);
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/ColorSupplier.java:7]: make a global Random field and use it here

File: src/main/java/core/basesyntax/Lottery.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import java.util.Random;
+
+public class Lottery {
+    public Ball getRandomBall() {
+        Ball randomBall = new Ball();
+        ColorSupplier colorSupplier = new ColorSupplier();
+        final int bound = 100;
+
+        randomBall.setColor(colorSupplier.getRandomColor());
+        randomBall.setNumber(new Random().nextInt(bound));
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Lottery.java:12]: make global Random and ColorSupplier fields and use them here

File: src/main/java/core/basesyntax/Lottery.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    public Ball getRandomBall() {
+        Ball randomBall = new Ball();
+        ColorSupplier colorSupplier = new ColorSupplier();
+        final int bound = 100;
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Lottery.java:9]: example of constant: private static final int DATE_INDEX = 0;

**Review by Nick97-git — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/ColorSupplier.java (hunk 2 above)
Comment (ValeriyM250) [reply src/main/java/core/basesyntax/ColorSupplier.java:7]: I apologize, but I didn`t understand what does it mean "global Random field and use it here"

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 5]
```diff
@@ -1,7 +1,20 @@
...
 public class ColorSupplier {
+    private Random random = new Random();
+
+    public Random getRandom() {
+        return random;
+    }
+
+    public void setRandom(Random random) {
+        this.random = random;
+    }
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/ColorSupplier.java:14]: I don't think we need this getters and setters

File: src/main/java/core/basesyntax/Lottery.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        this.randomBall = randomBall;
+    }
+
+    public ColorSupplier getColorSupplier() {
+        return colorSupplier;
+    }
+
+    public void setColorSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+    }
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/Lottery.java:28]:
```suggestion
public class Lottery {
    private static final int MAX_BALL_NUMBER = 100;
    private Random random = new Random();
    private ColorSupplier colorSupplier = new ColorSupplier();

    public Ball getRandomBall() {
        Ball ball = new Ball();
        ball.setColor(colorSupplier.getRandomColor());
        ball.setNumber(random.nextInt(MAX_BALL_NUMBER));
        return ball;
    }
```

File: src/main/java/core/basesyntax/Ball.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        this.color = color;
+    }
+
+    public int getNumber() {
+        return number;
+    }
+
+    public void setNumber(int number) {
+        this.number = number;
+    }
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/Ball.java:21]:
```suggestion
    public void setColor(String color) {
        this.color = color;
    }

    public void setNumber(int number) {
        this.number = number;
    }
```

do we use getters in our code?

**Review by MaliukDaria — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Lottery.java [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+import java.util.Random;
+
+public class Lottery {
+    private static final int MAX_BALL_NUMBER = 100;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Ball getRandomBall() {
+
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/Lottery.java:11]: redundant empty line

**Review by MaliukDaria — APPROVED** [review]

**Pull request closed without merging**